### PR TITLE
MON-130419-CMA_UI_Installer

### PR DESCRIPTION
**Pull request by jean-christophe81**
## Description

Centreon monitoring agent windows installer

REFS:MON-130419

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [X] master

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: .github/workflows/windows-agent.yml [hunk 1]
```diff
@@ -61,27 +68,10 @@ jobs:
...
-          enableCrossOsArchive: ${{ true }}
-
       - name: Upload package artifacts
-        if: |
-          github.event_name != 'workflow_dispatch' &&
-          contains(fromJson('["stable"]'), needs.get-version.outputs.stability) &&
-          ! cancelled() &&
-          ! contains(needs.*.result, 'failure') &&
-          ! contains(needs.*.result, 'cancelled')
+        if: inputs.installer_in_artifact == true
```
Comment (kduret) [inline .github/workflows/windows-agent.yml:72]: stable releases will not upload installer in artifact automatically, is it what you want ?

File: agent/installer/centreon-monitoring-agent-modify.nsi [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        ${If} $0 != "true"
+            ${Break}
+        ${EndIf}
+        Sleep 500
+    ${Loop}
+    ; even if service is stopped, process can be stopping so we wait a little more
+    Sleep 500
+
+    !insertmacro SERVICE "start" "${SERVICE_NAME}" ""
+    MessageBox MB_OK "Centreon Monitoring Agent is now restarted"
```
Comment (cg-tw) [inline agent/installer/centreon-monitoring-agent-modify.nsi:103]:
```suggestion
    MessageBox MB_OK "The Centreon Monitoring Agent has now restarted"
```

File: agent/installer/centreon-monitoring-agent.nsi [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+Function get_plugins_url
+    #because of several bugs, we have to store github response in a file and then parse it
+    Var /GLOBAL json_content_path
+    Var /GLOBAL nb_assets 
+    GetTempFileName $json_content_path
+    #get release information
+    ClearErrors
+    inetc::get /header "Accept: application/vnd.github+json" ${NSCLIENT_URL} $json_content_path /End
+    ${If} ${Errors}
+        MessageBox MB_OK|MB_ICONSTOP "Fail to get plugins informations from ${NSCLIENT_URL}"
```
Comment (cg-tw) [inline agent/installer/centreon-monitoring-agent.nsi:116]:
```suggestion
        MessageBox MB_OK|MB_ICONSTOP "Failed to get plugin information from ${NSCLIENT_URL}"
```

File: agent/installer/centreon-monitoring-agent.nsi [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+        Abort
+    ${EndIf}
+    Pop $0
+    ${If} $0 != "OK"
+        MessageBox MB_OK|MB_ICONSTOP "Fail to get plugins informations from ${NSCLIENT_URL}: $0"
```
Comment (cg-tw) [inline agent/installer/centreon-monitoring-agent.nsi:121]:
```suggestion
        MessageBox MB_OK|MB_ICONSTOP "Failed to get plugin information from ${NSCLIENT_URL}: $0"
```

File: agent/installer/dlg_helper.nsi [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+	${EndIf}
+
+    Pop $0
+FunctionEnd
+
+/**
+  * @brief hostname I image onClick handler
+*/
+Function hostname_help_onClick
+    MessageBox MB_ICONINFORMATION "This is the name of the host as defined in the Centreon interface."
```
Comment (cg-tw) [inline agent/installer/dlg_helper.nsi:116]:
```suggestion
    MessageBox MB_ICONINFORMATION "The name of the host as defined in the Centreon interface."
```

File: agent/installer/dlg_helper.nsi [hunk 6]
```diff
@@ -0,0 +1,452 @@
...
+        MessageBox MB_ICONINFORMATION "IP address of DNS name of the poller the agent will connect to."
+    ${EndIf}
+    Pop $0
+FunctionEnd
+
+/**
+  * @brief Poller-initiated connection checkbox I image onClick handler
+*/
+Function reverse_help_onClick
+    MessageBox MB_ICONINFORMATION "Use when the agent cannot connect to the poller directly: the poller will then initiate the connection."
```
Comment (cg-tw) [inline agent/installer/dlg_helper.nsi:137]:
```suggestion
    MessageBox MB_ICONINFORMATION "Use when the agent cannot connect to the poller directly: the poller will initiate the connection."
```

File: agent/installer/dlg_helper.nsi [hunk 7]
```diff
@@ -0,0 +1,452 @@
...
+        ShowWindow $hCtl_log_dlg_max_file_size_help ${SW_HIDE}
+    ${EndIf}
+
+FunctionEnd
+
+/**
+  * @brief max file I image onClick handler
+*/
+Function max_files_help_onClick
+    MessageBox MB_ICONINFORMATION "For the rotation of logs to be active, it is necessary that both parameters 'Max File Size' and 'Max number of files' are set. The space used by the logs of the agent will not exceed 'Max File Size' * 'Max Files'."
```
Comment (cg-tw) [inline agent/installer/dlg_helper.nsi:260]:
```suggestion
    MessageBox MB_ICONINFORMATION "For the rotation of logs to be active, it is necessary that both parameters 'Max File Size' and 'Max number of files' are set. The space used by the logs of the agent will not exceed 'Max File Size' * 'Max number of files'."
```

File: agent/installer/dlg_helper.nsi [hunk 8]
```diff
@@ -0,0 +1,452 @@
...
+FunctionEnd
+
+/**
+  * @brief max file size I image onClick handler
+*/
+Function max_file_size_help_onClick
+    MessageBox MB_ICONINFORMATION "For the rotation of logs to be active, it is necessary that both parameters 'Max File Size' and 'Max number of files' are set. The space used by the logs of the agent will not exceed 'Max File Size' * 'Max Files'."
```
Comment (cg-tw) [inline agent/installer/dlg_helper.nsi:267]:
```suggestion
    MessageBox MB_ICONINFORMATION "For the rotation of logs to be active, it is necessary that both parameters 'Max File Size' and 'Max number of files' are set. The space used by the logs of the agent will not exceed 'Max File Size' * 'Max number of files'."
```

File: agent/installer/dlg_helper.nsi [hunk 9]
```diff
@@ -0,0 +1,452 @@
...
+                Pop $1
+                Pop $0
+                Abort
+            ${EndIf}
+        ${EndIf}
+        Push $2
+        ${NSD_GetText} $hCtl_encryption_private_key_file_Txt $2
+        ${If} $2 == ""
+            ${If} $reverse_connection > 0
+                MessageBox MB_OK|MB_ICONSTOP "If encryption and poller-initiated connection are active, The private key is mandatory."
```
Comment (cg-tw) [inline agent/installer/dlg_helper.nsi:339]:
```suggestion
                MessageBox MB_OK|MB_ICONSTOP "If encryption and poller-initiated connection are active, the private key is mandatory."
```

File: agent/installer/dlg_helper.nsi [hunk 10]
```diff
@@ -0,0 +1,452 @@
...
+	${EndIf}
+    Pop $0
+FunctionEnd
+
+
+/**
+  * @brief private key file I image onClick handler
+*/
+Function private_key_file_help_onClick
+    MessageBox MB_ICONINFORMATION "Private key file path. Mandatory in case of reverse connection (poller to agent)."
```
Comment (cg-tw) [inline agent/installer/dlg_helper.nsi:430]:
```suggestion
    MessageBox MB_ICONINFORMATION "Private key file path. Mandatory if encryption and poller-initiated connection are active."
```

File: agent/installer/dlg_helper.nsi [hunk 11]
```diff
@@ -0,0 +1,452 @@
...
+FunctionEnd
+
+/**
+  * @brief certificate file I image onClick handler
+*/
+Function certificate_file_help_onClick
+    MessageBox MB_ICONINFORMATION "Public certificate file path. Mandatory in case of reverse connection (poller to agent)."
```
Comment (cg-tw) [inline agent/installer/dlg_helper.nsi:437]:
```suggestion
    MessageBox MB_ICONINFORMATION "Public certificate file path. Mandatory if encryption and poller-initiated connection are active."
```

**Review by kduret — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Review by bouda1 — APPROVED** [review]

**Pull request merged**